aten/src/ATen/native/cuda/ForeachBinaryOpList.cu

```diff
@@ -230,16 +230,18 @@ FOREACH_BINARY_OP_LIST(
     div,
     std::divides,
     /*division_op*/ true);
+// NOTE(crcrpar): `all_types_half_bfloat16` does not cover bool, so temporarily
+// set `division_op` to true.
```

Comment:
Wouldn't it be better to add another function that has bool? The division op being true for these is weird....

Reply:
I thought it would be redundant to have another macro with mostly same lines as the existing one

Reply:
i found it more natural to rename `division_op` to something else like `promote_int_to_float` as in https://github.com/pytorch/pytorch/blob/5033d3ba6dadbcf3bdeb5d0e76accf815e8039f1/aten/src/ATen/native/ForeachUtils.h#L234-L237

Reply:
oooooh i finally understand what this is + completely agree that the division_op name is bad. Could we document what the flag is ultimately used for? (that it is used as a condition when checking the op could go through the fast path. If it will convert ints or bools to floats, it is not supported on the fast path)

Maybe here or in _check_tensors_do_type_promotion_with_scalars